src/cargo/core/features.rs

```diff
@@ -1287,6 +1289,15 @@ impl CliUnstable {
             "profile-rustflags" => self.profile_rustflags = parse_empty(k, v)?,
             "trim-paths" => self.trim_paths = parse_empty(k, v)?,
             "publish-timeout" => self.publish_timeout = parse_empty(k, v)?,
+            "root-path" => self.root_path = v.map(|v| {
+                // Make the path absolute, if we can.
+                let v = PathBuf::from(v);
+                if v.is_absolute() {
+                    v
+                } else {
+                    v.canonicalize().unwrap_or(v)
+                }
```

Comment:
Cargo generally does not use `canonicalize` (suprised we haven't banned it with clippy).  We should use `gctx.cwd()`. 

We also don't generally do `is_absolute` checks but always do the join, maybe with a `cargo_util::paths::normalize_path` call.

Reply:
I was wondering about that since I recalled `canonicalize` behaving oddly on Windows... I thought I had seen a WIP new function for std that would behave better, but couldn't find it.

`normalize_path` is probably a good idea since we need the prefix to match exactly for `strip_prefix` to kick in.